matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/EventRelationsAggregationTask.kt

```diff
@@ -0,0 +1,309 @@
+                            }
+
+                            sum.addedByMe = sum.addedByMe || (userId == event.sender)
+                        }
+
+                    }
+                }
+
+            }
+        }
+    }
+
+    /**
+     * Called when an event is deleted
+     */
+    fun handleRedactionOfReplace(redacted: EventEntity, relatedEventId: String, realm: Realm) {
+        Timber.d("Handle redaction of m.replace")
+        val eventSummary = EventAnnotationsSummaryEntity.where(realm, relatedEventId).findFirst()
+        if (eventSummary == null) {
+            Timber.w("Redaction of a replace targeting an unknown event $relatedEventId")
+            return
+        }
+        val sourceEvents = eventSummary.editSummary?.sourceEvents
+        val sourceToDiscard = sourceEvents?.indexOf(redacted.eventId)
+        if (sourceToDiscard == null) {
+            Timber.w("Redaction of a replace that was not known in aggregation $sourceToDiscard")
+            return
+        }
+        //Need to remove this event from the redaction list and compute new aggregation state
+        sourceEvents.removeAt(sourceToDiscard)
+        val previousEdit = sourceEvents.mapNotNull { EventEntity.where(realm, it).findFirst() }.sortedBy { it.originServerTs }.lastOrNull()
+        if (previousEdit == null) {
+            //revert to original
+            eventSummary.editSummary?.deleteFromRealm()
+        } else {
+            //I have the last event
+            ContentMapper.map(previousEdit.content)?.toModel<MessageContent>()?.newContent?.let { newContent ->
+                eventSummary.editSummary?.lastEditTs = previousEdit.originServerTs
+                        ?: System.currentTimeMillis()
+                eventSummary.editSummary?.aggregatedContent = ContentMapper.map(newContent)
+            } ?: run {
+                Timber.e("Failed to udate edited summary")
+                //TODO how to reccover that
+            }
+
+        }
+    }
+
+    fun handleReactionRedact(eventToPrune: EventEntity, realm: Realm, userId: String) {
+        Timber.d("REDACTION of reaction ${eventToPrune.eventId}")
+        //delete a reaction, need to update the annotation summary if any
+        val reactionContent: ReactionContent = EventMapper.map(eventToPrune).content.toModel()
+                ?: return
+        val eventThatWasReacted = reactionContent.relatesTo?.eventId ?: return
+
+        val reactionkey = reactionContent.relatesTo.key
+        Timber.d("REMOVE reaction for key $reactionkey")
+        val summary = EventAnnotationsSummaryEntity.where(realm, eventThatWasReacted).findFirst()
+        if (summary != null) {
+            summary.reactionsSummary.where()
+                    .equalTo(ReactionAggregatedSummaryEntityFields.KEY, reactionkey)
+                    .findFirst()?.let { summary ->
+                        Timber.d("Find summary for key with  ${summary.sourceEvents.size} known reactions (count:${summary.count})")
```

Comment:
I do not use Timber.d anymore because it's not logged out on some device. Min level is Timber.v

Reply:
done